### ISSUE TITLE
Fields not generated for inline fragments

### ISSUE THREAD

**Issue by zcourts** [comigor/artemis#273]
**Before reporting a bug, please test the beta branch!**

## Bug description
Given a GraphQL like this:
```graphql
query ProductsData($arcql: String!)
{
    find(type:Product, arcql: $arcql){
        edges {
            node{
                ... on Product {
                    hypi {
                        id
                    }
                    title
                    description
                }
            }
        }
    }
}
```

The following Dart is generated:
```Dart
@JsonSerializable(explicitToJson: true)
class ProductsData$Query$Find$Edges with EquatableMixin {
  ProductsData$Query$Find$Edges();

  factory ProductsData$Query$Find$Edges.fromJson(Map<String, dynamic> json) =>
      _$ProductsData$Query$Find$EdgesFromJson(json);

  ProductsData$Query$Find$Edges$Node node;

  @override
  List<Object> get props => [node];
  Map<String, dynamic> toJson() => _$ProductsData$Query$Find$EdgesToJson(this);
}
```

It had to be changed to:
```graphql
query ProductsData($arcql: String!)
{
    find(type:Product, arcql: $arcql){
        edges {
            node{
                #must be a fragment - inline-fragments not supported
                ...product
            }
        }
    }
}

fragment product  on Product {
    __typename
    hypi {
        id
    }
    title
    description
}
```
Then it generated:
```Dart
@JsonSerializable(explicitToJson: true)
class ProductMixin$Hypi with EquatableMixin {
  ProductMixin$Hypi();

  factory ProductMixin$Hypi.fromJson(Map<String, dynamic> json) =>
      _$ProductMixin$HypiFromJson(json);

  String id;

  @override
  List<Object> get props => [id];
  Map<String, dynamic> toJson() => _$ProductMixin$HypiToJson(this);
}
...other generated stuff here then
    FragmentDefinitionNode(
        name: NameNode(value: 'product'),
        typeCondition: TypeConditionNode(
            on: NamedTypeNode(
                name: NameNode(value: 'Product'), isNonNull: false)),
        directives: [],
        selectionSet: SelectionSetNode(selections: [
          FieldNode(
              name: NameNode(value: '__typename'),
              alias: null,
              arguments: [],
              directives: [],
              selectionSet: null),
          FieldNode(
              name: NameNode(value: 'hypi'),
              alias: null,
              arguments: [],
              directives: [],
              selectionSet: SelectionSetNode(selections: [
                FieldNode(
                    name: NameNode(value: 'id'),
                    alias: null,
                    arguments: [],
                    directives: [],
                    selectionSet: null)
              ])),
          FieldNode(
              name: NameNode(value: 'title'),
              alias: null,
              arguments: [],
              directives: [],
              selectionSet: null),
          FieldNode(
              name: NameNode(value: 'description'),
              alias: null,
              arguments: [],
              directives: [],
              selectionSet: null)
        ]))
```

## Specs
Artemis version: 6.18.4
<details>
<summary>build.yaml:</summary>
  
```yaml
targets:
  $default:
    sources:
      - lib/**
      - graphql/**
      - schema.graphql
    builders:
      artemis:
        options:
          schema_mapping:
            - schema: schema.graphql
              queries_glob: graphql/*.graphql
              output: lib/models/graphql/graphql_api.dart
              naming_scheme: pathedWithFields
```
</details>
<details>

I don't have the other info being asked for in the issue template, I'll have to ask the dev working on it.
Are inline fragments suppose to work?

**Comment by zcourts** [comment]
It seems unions are not supported either.

**Comment by BreSmit521** [comment]
I feel like this is also related to some of the things I posed on #282 would be nice to have an acknowledgement on this one way or the other

**Comment by zcourts** [comment]
#282 appears to fix this in `v7.1.1-beta.1`
